syntax/verilog_systemverilog.vim

```diff
@@ -150,111 +150,67 @@ endif
-        \ end="\<endinterface\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement interface endinterface
-endif
-if index(s:verilog_syntax_fold, "clocking") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<clocking\>"
-        \ end="\<endclocking\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement clocking endclocking
-endif
-if index(s:verilog_syntax_fold, "covergroup") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<covergroup\>"
-        \ end="\<endgroup\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  covergroup endgroup
-endif
-if index(s:verilog_syntax_fold, "sequence") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<sequence\>"
-        \ end="\<endsequence\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  sequence endsequence
-endif
-if index(s:verilog_syntax_fold, "property") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<property\>"
-        \ end="\<endproperty\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  property endproperty
-endif
-if index(s:verilog_syntax_fold, "specify") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
-    syn region  verilogFold
-        \ matchgroup=verilogStatement
-        \ start="\<specify\>"
-        \ end="\<endspecify\>"
-        \ transparent
-        \ keepend
-        \ fold
-else
-    syn keyword verilogStatement  specify endspecify
-endif
-if index(s:verilog_syntax_fold, "block_nested") >= 0 || index(s:verilog_syntax_fold, "block_named") >= 0
+for name in ['class', 'clocking', 'covergroup', 'function', 'interface',
+    \ 'property', 'sequence', 'specify', 'task', ]
+
+    if name == 'task' || name == 'function'
+        let s:region_start = s:verilog_function_task_dequalifier.'\@<!\<'.name.'\>'
+    elseif name == 'class'
+        let s:region_start = '\(typedef\s\+\)\@<!\<\(interface\s\+\)\?class\>'
+    elseif name == 'interface'
+        let s:region_start = '\<interface\>\(\s\+class\)\@!'
+    else
+        let s:region_start = '\<'.name.'\>'
+    endif
+
+    if name == 'covergroup'
+        let s:region_end = '\<endgroup\>'
+    else
+        let s:region_end = '\<end'.name.'\>'
+    endif
+
+    if verilog_systemverilog#VariableExists('verilog_quick_syntax')
+        execute 'syn match verilogStatement "'.s:region_start.'"'
+        execute 'syn match verilogStatement "'.s:region_end.'"'
+    else
+        let s:verilog_syn_region_cmd =
+            \  'syn region verilog_'.name
```

Comment:
You forgot to revert to the old syntax name formats. Example: "verilogFunction" instead of "verilog_function".